st2common/st2common/util/virtualenvs.py

```diff
@@ -73,6 +74,19 @@ def setup_pack_virtualenv(pack_name, update=False, logger=None, include_pip=True
         msg = 'Pack "%s" is not installed. Looked in: %s' % (pack_name, search_paths)
         raise Exception(msg)
 
+    try:
+        pack_meta = get_pack_metadata(pack_path)
```

Comment:
Perhaps, for dependency injection and testing reasons, this function should take `use_python3` as an argument instead of reading the metadata file itself.

Reply:
Thinking out loud - I first thought this might slow down things quite a bit when running `st2ctl register --register-setup-virtualenvs` flag because we now also need to read and parse metadata file for each pack.

But after some more thought - we already need to do that, so another read and parse shouldn't add too much overhead since file should already be in filesystem cache. Future optimization would perhaps be to refactor the code a bit so we don't need to read and parse metadata file multiple times, but probably not needed right now.

Reply:
I assumed that the pack would still install if the meta-file wasn't in place, so I didn't want to alter the original behaviour

Reply:
Metadata is mandatory so it would fail at some other step, but that's not really important here and I'm fine with the current approach.

I was mostly just "thinking out loud" about the potential performance overhead from reading and parsing pack metadata multiple times during install phase :)